src/App.jsx

```diff
@@ -49,7 +50,43 @@ export function App() {
 			setTimeOutErrorMsg('New List Error. Please refresh or try again later.');
 		}
 	};
-
+	/**
+	 * Callback function gets passed as a prop through Home component to retrieve user input.
+	 * A Promise.race() method is used again as with setList to appropriately respond to the user if the database is not responding.
+	 * StreamListItems function takes the user submitted token and returns a snapshot of the items in the database.
+	 * If the collection is not empty, the shared token is set to localStorage.
+	 * Else the user is informed the token that was shared with them is empty or does not exist via setJoinListErrorMsg state that is passed through Home component.
+	 * **/
+	const joinList = async (token) => {
```

Comment:
Recycled existing setList sync function for the new feature. The handling was primarily the same, and the logic differentiated only in the first Promise.

Reply:
If the code is mostly the same, thoughts on putting it in a function and just calling that both place? (It is possible this won't work btw, as I notice a lot of things happening all in the same function)